src/controller/reducers/topicDashboard.js

```diff
@@ -0,0 +1,68 @@
+import {
+  RESOURCE_LOADING,
+  RESOURCE_SUCCESS,
+  ADD_RESOURCE,
+  UPDATE_RESOURCE,
+  DELETE_RESOURCE,
+  RESOURCE_ERROR
+} from '../actions/topicDashboard';
+
+import produce from 'immer';
+
+export const initialState = {
+  resources: [],
+  loading: false,
+  error: null,
+  topicID: null,
+  topicTitle: ''
+};
+
+export default produce((state, action) => {
+  switch (action.type) {
+    case RESOURCE_LOADING:
+      state.loading = true;
+      state.error = null;
+
+      return;
+
+    case RESOURCE_SUCCESS:
```

Comment:
I recommend using reducer setup for your initializing action instead:

```suggestion
case RESOURCE_SUCCESS:
  state.loading = false;
  state.error = null;

  // One thing that will make your life 1000000x easier is adding your resources
  // to an object using this function:
  function mapResourcesToObject(resources) {
    return resources.reduce((obj, resource) => {
      obj[resource.id] = resource;
      return obj;
    }, {})
  }

  state.resources = mapResourcesToObject(action.payload.resources);
  // remove redundant resources from payload
  delete action.payload.resources;

  state.resourceOrder = action.payload.resourceOrder;
  // remove redundant resourceOrder from payload
  delete action.payload.resourceOrder;

  state.topic = action.payload;
```

Here's what your state would look like if you adopted this idea:

```js
// EXAMPLE OF YOUR NEW STATE WITH THIS REDUCER

state.topicDashReducer = {
  topic: {
    "id": 4000,
    "title": "Node",
    "parent": {
        "id": 3001,
        "title": "Javascript"
    },
    "createdAt": "2018-12-13T23:39:04.638Z",
    "updatedAt": "2018-12-13T23:39:04.638Z",
  },
  resourceOrder: [1, 2, 3],
  resources: {
  "1": {
      "id": 1,
      "parent": {
          "id": 4000,
          "title": "Node"
      },
      "title": "Resource #1",
      "type": "youtube",
      "uri": "4DfzRRkiPX4",
      "completed": false,
      "lastOpened": "2018-12-12T22:28:18.573Z"
  },
  "2": {
      "id": 2,
      "parent": {
          "id": 4000,
          "title": "Node"
      },
      "title": "Resource #2",
      "type": "youtube",
      "uri": "B7hVxCmfPtM",
      "completed": false,
      "lastOpened": "2018-12-13T14:38:11.895Z"
  },
  "3": {
      "id": 3,
      "parent": {
          "id": 4001,
          "title": "ES6"
      },
      "title": "Resource #3",
      "type": "youtube",
      "uri": "B7hVxCmfPtM",
      "completed": false,
      "lastOpened": "2018-12-13T16:01:29.105Z"
  },
  loading: false,
  error: null
}
```

Notice how you can access your resources in an object now by using their IDs? This is 10x easier for your use case than trying to find it in an array for each and every item. The order is given by the IDs in the resourceOrder array, so you can quite easily load things in order by iterating through the resourceOrder array and look up the resources by ID.

FYI this is an extremely common thing to do. It's called data normalization and it makes your life a lot easier. [Here's a good article on it](https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape#designing-a-normalized-state).